menu.rb

```diff
@@ -0,0 +1,148 @@
+
+    # prints the menu
+    while count < number_of_menu_items
+
+      # removes a random object in the array
+      food1 = temp_food_adj.slice!(rand(temp_food_adj.length))
+      food2 = temp_food_loc.slice!(rand(temp_food_loc.length))
+      food3 = temp_food.slice!(rand(temp_food.length))
+
+      menu_item = "#{food1} #{food2} #{food3}"
+
+      # increases menu item count
+      count += 1
+
+      puts "#{count}. #{menu_item}"
+
+    end   # end of while loop printing the menu
+
+  # if the user wanted to make their own menu
+  elsif user_input == 2
+    puts "How many menu items would you like to add?"
+    user_input = gets.chomp.strip.to_i
+
+    # creates empty arrays for the user to fill in
+    user_food_adj = []
+    user_food_des = []
+    user_food = []
+
+    # prompts user for adjectives x (user_input) amount of times
+    count = 1
+    puts "Please provide #{user_input} adjective(s).\n(Press ENTER after every adjective.)"
+    while count <= user_input
+      print "#{count}. "
+      user_food_adj[count - 1] = gets.chomp
+      count += 1
+    end
+
+    # prompts user for discriptors x (user_input) amount of times
+    count = 1
+    puts "Please provide #{user_input} discriptor(s) (how cooked, origin, color, etc).\n(Press ENTER after every descriptor.)"
+    while count <= user_input
+      print "#{count}. "
+      user_food_des[count - 1] = gets.chomp
+      count += 1
+    end
+
+    # prompts user for food x (user_input) amount of times
+    count = 1
+    puts "Please provide #{user_input} food item(s).\n(Press ENTER after each food entry.)"
+    while count <= user_input
+      print "#{count}. "
+      user_food[count - 1] = gets.chomp
+      count += 1
+    end
+
+    puts "Are you ready for your custom menu? WELL HERE IT IS:"
+    count = 0   # for the menu numbering
+
+    # prints the menu (this is not optimized--copy and pasted from above)
+    while count < user_input
+
+      # removes a random object in the array
+      food1 = user_food_adj.slice!(rand(user_food_adj.length))
+      food2 = user_food_des.slice!(rand(user_food_des.length))
+      food3 = user_food.slice!(rand(user_food.length))
+
+      menu_item = "#{food1} #{food2} #{food3}"
+
+      # increases menu item count
+      count += 1
+
+      puts "#{count}. #{menu_item}"
+
+    end   # end of while loop printing the menu
+
+  else   # should never trigger
+    puts "I don't know how you got here, you shouldn't be here."
+    break
+  end   # end of if-elsif-else conditional (user menu choice)
+
+  # asks user if they want to repeat the program
+  puts "Would you like to re-enter Food World? Y / N"
+  user_input = gets.chomp.upcase.strip
+
+  # loop repeats until 'Y' or 'N' are entered by the user
+  until (user_input == "Y") || (user_input == "N")
+    puts "I only understand single-letter responses. Y / N?"
+    user_input = gets.chomp.upcase.strip
+  end
+
+  if (user_input == "N")
+    break   # breaks out of the while loop
```

Comment:
I was a bit worried about using a `while true` loop at the start, but I see what you're doing here. I'd just be very _very_ careful when using this type of loop to be sure you don't end up in an infinite loop.